### PR TITLE
56532 add notice

### PR DESCRIPTION
**Pull request by robinwpdeveloper**
Security notice added for 4.0 or older version.

Trac ticket: [https://core.trac.wordpress.org/ticket/56532](https://core.trac.wordpress.org/ticket/56532)

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/wp-admin/update-core.php [hunk 1]
```diff
@@ -269,6 +269,17 @@ function core_upgrade_preamble() {
 			__( 'https://wordpress.org/support/article/updating-wordpress/' )
 		);
 		echo '</p></div>';
+
+		if ( ! empty( $wp_version ) && $wp_version <= 4.0 ) {
```
Comment (rudlinkon) [inline src/wp-admin/update-core.php]:
```suggestion
		if ( version_compare( $wp_version, '4.1.0', '<' ) ) {
```

**Review by rudlinkon — CHANGES_REQUESTED** [review]
it is safe to use `version_compare` function and no need to check if is it empty because this variable is already used in the upper block.

**Comment by robinwpdeveloper** [comment]
Thanks @rudlinkon . Requested changes are made and pushed.

File: src/wp-admin/update-core.php [hunk 2]
```diff
@@ -269,6 +269,17 @@ function core_upgrade_preamble() {
 			__( 'https://wordpress.org/support/article/updating-wordpress/' )
 		);
 		echo '</p></div>';
+
+		if ( version_compare( $wp_version, '4.1.0', '<' ) ) {
+			echo '<div class="notice notice-warning inline"><p>';
+			printf(
+				/* translators: 1: Documentation on WordPress backups, 2: Documentation on updating WordPress. */
```
Comment (mukeshpanchal27) [inline src/wp-admin/update-core.php]: It should be WordPress version.

```suggestion
				/* translators: 1: WordPress version, 2: Documentation on updating WordPress. */
```

**Review by mukeshpanchal27 — COMMENTED** [review]
@robinwpdeveloper Left feedback for translator document.

**Comment by robinwpdeveloper** [comment]
Thanks @mukeshpanchal27 . Code updated.

**Comment by qasumitbagthariya** [comment]
I have tested this PR in WordPress 3.9.37 and it is working as expected, Please note I have manually put the PR code in the `src/wp-admin/update-core.php` file. Please check the video.

**Environment**
OS: Windows 10
Web Server: nginx/1.16.5
PHP: 7.4.27 
WordPress: 3.9.37
Theme: Twentytwentytwo

**Browsers:**
Chrome - Version 105.0.5195.127 (Official Build) (64-bit)


https://user-images.githubusercontent.com/67687255/191986979-8664227c-7599-4bfa-a534-a23425aca816.mp4

image
image

**Comment by rudlinkon** [comment]
> I have tested this PR in WordPress 3.9.37 and it is working as expected, Please note I have manually put the PR code in the `src/wp-admin/update-core.php` file. Please check the video.
> 
> **Environment** OS: Windows 10 Web Server: nginx/1.16.5 PHP: 7.4.27 WordPress: 3.9.37 Theme: Twentytwentytwo
> 
> **Browsers:** Chrome - Version 105.0.5195.127 (Official Build) (64-bit)
> 
>  56532.add.notice.by.robinwpdeveloper.Pull.Request.3213.WordPress_wordpress-develop.-.Google.Chrome.-.23.September.2022.mp4 
> image image

Thank you @qasumitbagthariya for testing the PR

**Comment by peterwilsoncc** [comment]
As mentioned on trac, for versions trunk and versions 4.0 and above the strings can be added to the collection of release related strings after the `return` statement.

This will allow the translation team to begin translating them without the need for additional code wrapped in the conditional.

https://github.com/WordPress/wordpress-develop/blob/8f3254d794ca4e8b8e0ec8254e5886be840a60c6/src/wp-admin/about.php#L342-L378

There are two strings that need to be  added to the about page:

```php
/* translators: 1: WordPress version number, 2: Link to update WordPress */
__( 'Important! Your version of WordPress (%1$s) is no longer supported, you will not receive any security updates for your website. To keep your site secure, please <a href="%2$s">update to the latest version of WordPress</a>.' );

/* translators: 1: WordPress version number, 2: Link to update WordPress */
__( 'Important! Your version of WordPress (%1$s) will stop receiving security updates in the near future. To keep your site secure, please <a href="%2$s">update to the latest version of WordPress</a>.' );
```

On the ticket, I've uploaded some patches for WordPress 3.7 - 4.0 that include the CSS changes required to make the update notice more prominent.

File: src/wp-admin/update-core.php [hunk 3]
```diff
@@ -269,6 +269,17 @@ function core_upgrade_preamble() {
...
+		if ( version_compare( $wp_version, '4.1.0', '<' ) ) {
+			echo '<div class="notice notice-warning inline"><p>';
+			printf(
+				/* translators: 1: WordPress version, 2: Documentation on updating WordPress. */
+				__( '<strong>Important:</strong> Your version of WordPress (%1$s) will stop receiving security updates in the near future. To keep your site secure, your site should use WordPress 4.1 or later. However, it\'s recommended to <a href="%2$s">update to the latest version</a> of WordPress.' ),
+				$wp_version,
+				__( 'https://wordpress.org/support/article/updating-wordpress/' )
+			);
+			echo '</p></div>';
+		}
```
Comment (peterwilsoncc) [inline src/wp-admin/update-core.php]:
```suggestion
```

**Review by peterwilsoncc — COMMENTED** [review]
The strings have been added correctly in the about page, thank you!

Are you able to remove the changes to `update-core.php` as they won't be needed for the current versions of WordPress for some time. I have some separate patches on trac for the versions of WP that will shortly be losing support

**Comment by robinwpdeveloper** [comment]
Thanks @peterwilsoncc 
Code updated as per your suggestion.

**Review by peterwilsoncc — APPROVED** [review]
Thanks so much, I think this is ready.

**Comment by peterwilsoncc** [comment]
Merged in https://core.trac.wordpress.org/changeset/54322 / 1de72e40208fcf45e44b26ad79a1fd3322ca1a2e

**Pull request closed without merging**